demo-circuit/src/naive_threshold_sig/mod.rs

```diff
@@ -117,19 +119,31 @@ impl<F: PrimeField> NaiveTresholdSignature<F> {
 
         //Compute cert_data_hash
         let cert_data_hash = {
+
+            // Compute wcert_sysdata_hash
             let wcert_sysdata_hash = FieldHash::init_constant_length(6, None)
                 .update(sc_id)
                 .update(epoch_number)
                 .update(mr_bt)
-                .update(valid_signatures_fe)
+                .update(quality)
                 .update(end_cumulative_sc_tx_comm_tree_root)
                 .update(fees_field_elements)
                 .finalize()
                 .unwrap();
-            FieldHash::init_constant_length(1, None)
-                .update(wcert_sysdata_hash)
-                .finalize()
-                .unwrap()
+            
+            // Compute custom_fields hash taking into account the presence, or not, of custom_fields
+            let to_hash = if let Some(custom_fields) = custom_fields.clone() {
+                let mut h = FieldHash::init_constant_length(custom_fields.len(), None);
+                custom_fields.into_iter().for_each(|custom_field| { h.update(custom_field); });
+                let custom_fields_hash = h.finalize().unwrap();
+                vec![custom_fields_hash, wcert_sysdata_hash]
```

Comment:
Why `custom_fields_hash` is not a part of `wcert_sysdata_hash`?
Is it fine and consistent with MC (if any relation)?

Reply:
This reflects exactly the computation performed in cctp-lib (https://github.com/HorizenOfficial/zendoo-cctp-lib/blob/dev/cctp_primitives/src/proving_system/verifier/certificate.rs#L11), so for sure mc-cryptolib and zendoo are aligned in the same way.
We don't have any certificate struct Rust side so it doesn't really matter at the end of the day

Reply:
great, thanks!